lib/cfg-grammar.y

```diff
@@ -903,7 +903,7 @@ template_content_inner
           CHECK_ERROR_GERROR(log_template_compile(last_template, $3, &error), @3, error, "Error compiling template");
           free($3);
 
-          CHECK_ERROR_GERROR(log_template_set_type_hint(last_template, $1, &error), @1, error, "Error setting the template type-hint");
+          CHECK_ERROR_GERROR(log_template_set_type_hint(last_template, $1, &error), @1, error, "Error setting the template type-hint \"%s\"", $1);
```

Comment:
I like the new error messages. Checked each use case. I think only this one will give a "double" error message, since the `error` string will also contain the original parameter: https://github.com/syslog-ng/syslog-ng/blob/master/lib/type-hinting.c#L70